tests/tests_fabric/utilities/test_spike.py

```diff
@@ -0,0 +1,65 @@
+import contextlib
+import sys
+from functools import partial
+
+import pytest
+import torch
+
+from lightning.fabric import Fabric
+from lightning.fabric.utilities.spike import _TORCHMETRICS_GREATER_EQUAL_1_0_0, SpikeDetection, TrainingSpikeException
+
+
+def spike_detection_test(fabric, global_rank_spike, spike_value, should_raise):
+    loss_vals = [1 / i for i in range(1, 10)]
+    if fabric.global_rank == global_rank_spike:
+        if spike_value is None:
+            loss_vals[4] = 3
+        else:
+            spike_value = spike_value
+
+    for i in range(len(loss_vals)):
+        context = pytest.raises(TrainingSpikeException) if i == 4 and should_raise else contextlib.nullcontext()
+
+        with context:
```

Comment:
Not sure, but I think the problem here is that you use multi-processing but only raise the exception in one rank. So your test will probably fail. Shouldn't the the `fabric.launch()` be wrapped with a try catch too in this case?

Reply:
no, the decision is synced in https://github.com/Lightning-AI/lightning/pull/18014/files#diff-6c07d896784502a485056fb611d5252a68c3dfa6b3eb8874444a4202750d1424R80 so both ranks raise the Exception :)

Reply:
Okay, I missed that